pkgs/tools/virtualization/cloud-init/0001-add-nixos-support.patch

```diff
@@ -72,7 +95,7 @@ diff -ruN cloud-init-0.7.6.orig/cloudinit/distros/nixos.py cloud-init-0.7.6/clou
 +        if not conf:
 +            conf = HostnameConf('')
 +        conf.set_hostname(your_hostname)
-+        util.write_file(out_fn, str(conf), 0644)
```

Comment:
is 0644 the default mode if left unspecified?

Reply:
It will use the permission of the old file, which is `.r--r--r-- 14 root  1 Jan  1970  /nix/store/g1mkk8lygngbzvm632wbmnm8xxf9dymz-etc-hostname
`

Reply:
Same here. The actual line you are interested in is below.